homeassistant/components/energyzero/sensor.py

```diff
@@ -0,0 +1,196 @@
+        key="lowest_price_time",
+        name="Time of lowest price - today",
+        service_type="today_energy",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data.energy_today.lowest_price_time,
+    ),
+    EnergyZeroSensorEntityDescription(
+        key="percentage_of_max",
+        name="Current percentage of highest price - today",
+        service_type="today_energy",
+        native_unit_of_measurement=PERCENTAGE,
+        icon="mdi:percent",
+        value_fn=lambda data: data.energy_today.pct_of_max_price,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up EnergyZero Sensors based on a config entry."""
+    coordinator: EnergyZeroDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        EnergyZeroSensorEntity(
+            coordinator=coordinator,
+            description=description,
+        )
+        for description in SENSORS
+    )
+
+
+class EnergyZeroSensorEntity(
+    CoordinatorEntity[EnergyZeroDataUpdateCoordinator], SensorEntity
+):
+    """Defines a EnergyZero sensor."""
+
+    _attr_has_entity_name = True
+    _attr_attribution = "Data provided by EnergyZero"
+    entity_description: EnergyZeroSensorEntityDescription
+
+    def __init__(
+        self,
+        *,
+        coordinator: EnergyZeroDataUpdateCoordinator,
+        description: EnergyZeroSensorEntityDescription,
+    ) -> None:
+        """Initialize EnergyZero sensor."""
+        super().__init__(coordinator=coordinator)
+        self.entity_description = description
+        self.entity_id = (
+            f"{SENSOR_DOMAIN}.{DOMAIN}_{description.service_type}_{description.key}"
+        )
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}_{description.service_type}_{description.key}"
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={
+                (
+                    DOMAIN,
+                    f"{coordinator.config_entry.entry_id}_{description.service_type}",
+                )
+            },
+            manufacturer="EnergyZero",
+            name=SERVICE_TYPE_DEVICE_NAMES[self.entity_description.service_type],
+        )
+
+    @property
+    def native_value(self) -> float | datetime | None:
+        """Return the state of the sensor."""
+        return self.entity_description.value_fn(self.coordinator.data)
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return the state attributes."""
+        if self.entity_description.key == "average_price":
+            return {
+                "today": self.coordinator.data.energy_today.timestamp_prices,
+                "tomorrow": self.coordinator.data.energy_tomorrow.timestamp_prices
+                if self.coordinator.data.energy_tomorrow
+                else None,
+            }
```

Comment:
This generates a great amount of data in state attributes, which is not what they are for.

![image](https://user-images.githubusercontent.com/195327/210008226-84e4346e-8ca3-4fdc-a637-6021c78ee41d.png)

We have discussed this with the core team, and came to this conclusion:

> It's too much data for state attributes.
> We don't accept that feature right now.
> We have plans for a solution but it's not ready yet.

These attributes need therefore be removed.

Reply:
Removed the extra attributes